### PR TITLE
Fixed sub_block definition bug and removed lbin PATH warning

### PR DESCRIPTION
**Pull request by ginty**
A subtle bug (breaking change) introduced by the recent `sub_block_groups` update is that a sub_block definition stopped overriding an existing method of the same name.

This restores the original behavior but adds a warning that this is happening.
Added a similar test case for the `sub_block_groups` method.

Also made a small update to allow `sub_block[:my_block]` (singular) to work, it would have previously crashed.

Finally, removed the no lbin in PATH warning discussed in #329.

### CODE REVIEW

**Comment by redxeth** [comment]
@ginty Thanks for this fix and adding the test case.

**Comment by ginty** [comment]
@redxeth, no problem.
Man, I totally noticed that the sub_blocks_group update had added in an `unless_defined` gate around the singleton method definition, but I thought it would be OK.
Turns out it broke an app, just shows how seriously we need to be taking any change of behavior now.

File: spec/sub_block_spec.rb [hunk 1]
```diff
@@ -538,6 +538,40 @@ def initialize
...
+          m = SB1.new
+          m.blah.should == 1
+          m.sub_block_group :blah do
+            m.sub_block :my_blah
+          end
+          m.blah.should_not == 1
+        end
+      end
+
+      it "adding a sub_block should not override an existing method of that name" do
```
Comment (redxeth) [inline spec/sub_block_spec.rb]: @ginty Should this read 'adding a sub_block should override...', i.e. remove the 'not' ?

**Comment by redxeth** [comment]
@ginty generally we need to be sure to always have a test case for any expected operation like that.   Currently running into this with the 'test_ids' plugin...

**Comment by info-rchitect** [comment]
@redxeth What issue with test_ids?

**Comment by redxeth** [comment]
@info-rchitect let me post an issue on that github page.   We're seeing some issues with next bin from range.  I fixed a bug causing program generation to take way too long as well, but it's not yet been merged as I'm working to add a good test case.

File: lib/origen/sub_blocks.rb [hunk 2]
```diff
@@ -295,10 +297,12 @@ def sub_block(name, options = {})
         if sub_blocks[name] && !sub_blocks[name].is_a?(Placeholder)
           fail "You have already defined a sub-block named #{name} within class #{self.class}"
         end
-        unless respond_to?(name)
-          define_singleton_method name do
-            get_sub_block(name)
-          end
+        if respond_to?(name)
+          callers = caller[0].split(':')
+          Origen.log.warning "The sub_block defined at #{Pathname.new(callers[0]).relative_path_from(Pathname.pwd)}:#{callers[1]} is overridding and existing method called #{name}"
```
Comment (pderouen) [inline lib/origen/sub_blocks.rb]: A small typo here. Should read, "overriding an ..."

**Pull request merged**